openedx/features/qverse_features/registration/validators.py

```diff
@@ -1,34 +1,64 @@
 """
 Validation utils for qverse registration application.
 """
+from csv import reader
+import io
+
 from django.core.exceptions import ValidationError
 
 
-def validate_file_extension(file):
+def validate_admission_file(file):
     """
-    Validates that file is a valid csv file by checking its extension.
+    Validates that the file is a valid csv file by checking its extension.
+
+    Also checks that the header row has all the required fields.
+
+    Arguments:
+        file (models.FileField): The file uploaded by the user
     """
     if not file.name.endswith('.csv'):
         raise ValidationError('', code='invalid')
 
+    FIELD_NAMES = [
+                'regno', 'firstname', 'surname', 'othername', 'levelid',
+                'programmeid', 'departmentid', 'mobile', 'email', 'mobile'
+                ]
+    decoded_file = file.read().decode('utf-8')
+    io_string = io.StringIO(decoded_file)
+    header_row = []
+    try:
+        header_row = reader(io_string, delimiter=',').next()
```

Comment:
the default delimiter is `,` so no need to specify this. Even I think we should take care of at least `\t` as well. Other popular delimiters include the tab (\t), colon (:) and semi-colon (;) characters.

Reply:
done.